### ISSUE TITLE
There still seem to be errors in the writing of the book.

### ISSUE THREAD

**Issue by fadfhkdfga** [ds58/Panilla#160]
![image](https://github.com/user-attachments/assets/c3825e4e-4c29-42f8-915c-432ccb05a83c)

An error that occurs when you first write a book or obtain a written book item.

**Comment by Folas1337** [comment]
I can confirm, this is also happening on my server :)

**Comment by dasavick** [comment]
Full exception

```java
[17:44:05 ERROR]: Could not pass event PlayerJoinEvent to Panilla v1.12.1
java.lang.ClassCastException: class com.ruinscraft.panilla.lib.nbtapi.NBTListCompound cannot be cast to class java.lang.String (com.ruinscraft.panilla.lib.nbtapi.NBTListCompound is in unnamed module of loader 'external-panilla.jar' @1536b219; java.lang.String is in module java.base of loader 'bootstrap')
        at external-panilla.jar/com.ruinscraft.panilla.paper.v1_21.nbt.NbtTagList.getString(NbtTagList.java:23) ~[external-panilla.jar:?]
        at external-panilla.jar/com.ruinscraft.panilla.api.nbt.checks.paper1_20_6.NbtCheck_WrittenBookContent.check(NbtCheck_WrittenBookContent.java:68) ~[external-panilla.jar:?]
        at external-panilla.jar/com.ruinscraft.panilla.api.nbt.checks.NbtChecks.checkAll(NbtChecks.java:172) ~[external-panilla.jar:?]
        at external-panilla.jar/com.ruinscraft.panilla.paper.v1_21.InventoryCleaner.clean(InventoryCleaner.java:42) ~[external-panilla.jar:?]
        at external-panilla.jar/com.ruinscraft.panilla.bukkit.JoinQuitListener.onJoin(JoinQuitListener.java:31) ~[external-panilla.jar:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor300.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-mojangapi-1.21.1-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-mojangapi-1.21.1-R0.1-SNAPSHOT.jar:1.21.1-2289-201e928]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-mojangapi-1.21.1-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:630) ~[paper-mojangapi-1.21.1-R0.1-SNAPSHOT.jar:?]
        at net.minecraft.server.players.PlayerList.placeNewPlayer(PlayerList.java:347) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.network.ServerConfigurationPacketListenerImpl.handleConfigurationFinished(ServerConfigurationPacketListenerImpl.java:173) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.network.protocol.configuration.ServerboundFinishConfigurationPacket.handle(ServerboundFinishConfigurationPacket.java:22) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.network.protocol.configuration.ServerboundFinishConfigurationPacket.handle(ServerboundFinishConfigurationPacket.java:13) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:56) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:151) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1574) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:201) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:125) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1551) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1544) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:135) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.managedBlock(MinecraftServer.java:1503) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1510) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1355) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:334) ~[purpur-1.21.1.jar:1.21.1-2289-201e928]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

**Comment by ds58** [comment]
Should be fixed in `1.12.3`
